### ISSUE TITLE
Avoid unnecessary memory rentention in recycled builder

### ISSUE THREAD

**Issue by jbgi** [h908714124/zerobuilder#1]
btw, thanks for this cool project!

Eg. the generated impl. of the builder for [Request#put](/h908714124/zerobuilder/blob/master/examples/derive4j/src/main/java/net/zerobuilder/examples/derive4j/Request.java#L33) is currently:

``` java
  static final class PutBuilderImpl implements PutBuilder.Path, PutBuilder.Body {
    private String path;
    private PutBuilderImpl() { }
    [...]
    @Override
    public Request body(String body) {
      return Request.put(path, body);
    }
  }
```

To avoid unnecessary memory retention if `recycled` is on, the implementation should nullify all instance fields before returning the target instance. I.e.:

``` java
  private static final class PutBuilderImpl implements PutBuilder.Path, PutBuilder.Body {
    private String path;
    PutBuilderImpl() { }
    [...]
    @Override
    public Request body(String body) {
      String path = this.path;
      this.path = null;
      return Request.put(path, body);
    }
  }
```

Also, on a minor note, I suggest moving the `private` modifier from the constructor onto the class declaration (as in my snippet). So that javac does not need to generate of a package-private synthetic constructor.

**Comment by h908714124** [comment]
Thanks @jbgi both suggestions seem valid and I will implement as soon as the API gets out of the current refactoring spree (as you can see I'm having fun with the visitor pattern...) BTW I learned this powerful pattern from derive4j, so thanks for that too!

**Comment by h908714124** [comment]
Both suggestions are implemented in `1.512`.
